### PR TITLE
Deflaked lb_policies_test

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

File: test/core/client_channel/lb_policies_test.c [hunk 1]
```diff
@@ -780,15 +782,18 @@ static void verify_total_carnage_round_robin(const servers_fixture *f,
...
-  /* no server is ever available. The persistent state is TRANSIENT_FAILURE */
+  /* no server is ever available. The persistent state is TRANSIENT_FAILURE. May
+   * also be CONNECTING if, under load, this check took too long to run and some
+   * subchannel already transitioned to retrying. */
   for (size_t i = 0; i < sequences->n; i++) {
     const grpc_connectivity_state actual = sequences->connectivity_states[i];
-    const grpc_connectivity_state expected = GRPC_CHANNEL_TRANSIENT_FAILURE;
-    if (actual != expected) {
+    const uint32_t expected_bitset =
+        GRPC_CHANNEL_TRANSIENT_FAILURE | GRPC_CHANNEL_CONNECTING;
```
Comment (markdroth) [inline test/core/client_channel/lb_policies_test.c]: The enum values are not bits, they're sequential values (i.e., the values are not 1, 2, 4, 8, etc; they are 0, 1, 2, 3, etc).  I don't think you can actually 'or' them together this way.  I think you actually need to check whether actual is equal to one or the other.

File: test/core/client_channel/lb_policies_test.c [hunk 2]
```diff
@@ -838,15 +842,21 @@ static void verify_partial_carnage_round_robin(
...
-  expected = GRPC_CHANNEL_TRANSIENT_FAILURE;
-  if (actual != expected) {
-    gpr_log(GPR_ERROR,
-            "CONNECTIVITY STATUS SEQUENCE FAILURE: expected '%s', got '%s' "
-            "at iteration #%d",
-            grpc_connectivity_state_name(expected),
-            grpc_connectivity_state_name(actual), (int)num_iters - 1);
-    abort();
+  const uint32_t expected_bitset =
+      GRPC_CHANNEL_TRANSIENT_FAILURE | GRPC_CHANNEL_CONNECTING;
```
Comment (markdroth) [inline test/core/client_channel/lb_policies_test.c]: Same here.

**Comment by dgquintas** [comment]
Ugh, the shame... Seeing all capitals tripped my brain :/ thanks!

**Comment by dgquintas** [comment]
Fixed. PTAL.

File: test/core/client_channel/lb_policies_test.c [hunk 3]
```diff
@@ -780,15 +783,21 @@ static void verify_total_carnage_round_robin(const servers_fixture *f,
...
 
-  /* no server is ever available. The persistent state is TRANSIENT_FAILURE */
+  /* no server is ever available. The persistent state is TRANSIENT_FAILURE. May
+   * also be CONNECTING if, under load, this check took too long to run and some
+   * subchannel already transitioned to retrying. */
   for (size_t i = 0; i < sequences->n; i++) {
     const grpc_connectivity_state actual = sequences->connectivity_states[i];
-    const grpc_connectivity_state expected = GRPC_CHANNEL_TRANSIENT_FAILURE;
-    if (actual != expected) {
+    uint32_t expected_bitset;
```
Comment (markdroth) [inline test/core/client_channel/lb_policies_test.c]: The bitsets really seem like overkill here.  Why not just say `if (actual != GRPC_CHANNEL_TRANSIENT_FAILURE && actual != GRPC_CHANNEL_CONNECTING)`?  I think it's both shorter and easier to understand.

Comment (dgquintas) [inline test/core/client_channel/lb_policies_test.c]: Must demonstrate complexity :) JK, fair point. Done.

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request merged**